src/commands.rs

```diff
@@ -475,7 +485,16 @@ where
                         }
                         _ => {
                             //TODO: something better here?
-                            return Err(e).context("error reading compile response from server");
+                            eprintln!("unexpected io error: {:?}", e.downcast_ref::<io::Error>());
```

Comment:
Since you're already `eprintln!()`-ing/returning `Err(e)`, this `eprintln!()` is probably redundant.